File: Examples/CSharp/ProgrammersGuide/Working-Diagrams/ExportToXPS.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Diagram. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Diagram;

namespace CSharp.Diagrams
{
    public class ExportToXPS
    {
        public static void Run()
        {
            //ExStart:ExportToXPS
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Diagrams();

            //Open VSD file
            Diagram diagram = new Diagram(dataDir + "ExportToXPS.vsd");

            //Save diagram to XPS format
            diagram.Save(dataDir + "Output.xps", SaveFileFormat.XPS);
            //ExEnd:ExportToXPS
        }
    }
}

File: Examples/VisualBasic/ProgrammersGuide/Load-Save-Convert/ExportToXPS.vb
'////////////////////////////////////////////////////////////////////////
' Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
'
' This file is part of Aspose.Diagram. The source code in this file
' is only intended as a supplement to the documentation, and is provided
' "as is", without warranty of any kind, either expressed or implied.
'////////////////////////////////////////////////////////////////////////

Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.Diagram

Namespace VisualBasic.Diagrams
    Public Class ExportToXPS
        Public Shared Sub Run()
            'ExStart:ExportToXPS
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_LoadSaveConvert()

            ' open VSD diagram
            Dim diagram As New Diagram(dataDir & "ExportToXPS.vsd")

            ' save diagram to XPS format
            diagram.Save(dataDir & "Output.xps", SaveFileFormat.XPS)
            'ExEnd:ExportToXPS
        End Sub
    End Class
End Namespace